src/server/hset_family_test.cc

```diff
@@ -203,6 +203,79 @@ TEST_F(HSetFamilyTest, HRandFloat) {
   Run({"hrandfield", "k"});
 }
 
+TEST_F(HSetFamilyTest, HRandFieldCount) {
+  // Setting keys
+  Run({"HSET", "a", "field1", "value1", "field2", "value2", "field3", "value3"});
+  Run({"HSET", "b", "field1", "value1"});
+
+  // Check validity of the keys
+  auto resp = Run({"hrandfield", "a", "3"});
+  EXPECT_THAT(resp, ArrLen(3));
+  bool unexpected_response_key = false, has_value = false;
+  set<string> unique_keys;
+  for (size_t i = 0; i < resp.GetVec().size(); i++) {
+    unique_keys.insert(resp.GetVec()[i].GetString());
+    if (resp.GetVec()[i] != "field1" && resp.GetVec()[i] != "field2" &&
+        resp.GetVec()[i] != "field3") {
+      unexpected_response_key = true;
+    }
```

Comment:
You can just make this an EXPECT instead of setting a variable. This is preferred because you know on what iteration the assert failed

There is also a gtest matcher AnyOf or smth like that so you don't have to type out the condition